contracts/PillarStaking.sol

```diff
@@ -162,35 +155,24 @@ contract PillarStaking is ReentrancyGuard, Ownable {
         return stakeholderData[account].stakedAmount;
     }
 
-    function getRewardAmountForAccount(address account)
-        external
-        view
-        returns (uint256 amount)
-    {
-        return stakeholderData[account].rewardAmount;
-    }
-
     /* ========== RESTRICTED FUNCTIONS ========== */
 
     function depositRewards(uint256 _amount) external onlyOwner {
         if (_amount == 0) revert RewardsCannotBeZero();
         rewardToken.safeTransferFrom(msg.sender, address(this), _amount);
-        rewardsDeposited = true;
+        rewards = _amount;
         emit RewardsDeposited(_amount);
     }
 
-    function calculateRewardAllocation() external onlyOwner {
-        if (!rewardsDeposited) revert RewardsNotTransferred();
-        uint256 totalRewards = rewardToken.balanceOf(address(this));
-        for (uint256 i; i < stakeholderList.length; ++i) {
-            address stakerAddress = stakeholderList[i];
-            uint256 stakeAmount = getStakedAmountForAccount(stakerAddress);
-            uint256 stakedPercentage = ((stakeAmount * BPS) / totalStaked);
-            uint256 reward = ((stakedPercentage * totalRewards) / BPS);
-            stakeholderData[stakerAddress].rewardAmount = reward;
-        }
-        rewardsAllocated = true;
-        emit RewardsAllocated();
+    function calculateRewardAllocation(address _staker)
```

Comment:
can we make this function `external whenReadyForUnstake` so users will be able to see their rewards?

Reply:
And remove it from inside of the unstake fn?

Reply:
ah, right, I meant `public whenReadyForUnstake`

Reply:
actually, if it's not the requirement, then let's skip it @lbw33

Reply:
No I think that it is a good idea. It would be good for users to see their reward allocation before they unstake.

Reply:
Done